### ISSUE TITLE
api key

### ISSUE THREAD

**Issue by fettahpinar** [gexgd0419/NaturalVoiceSAPIAdapter#17]
Hello.
I downloaded the latest update to my computer.
I got an api key.
I entered the api key and the region and saved it.
the azure sounds appear among the sapi 5 sounds, but it does not synthesize.
when i enter the same api key in balabolkaya and make it synthesize, it synthesizes without any problem.
also there was only edge sound list in the app data folder, I deleted the sound list manually.
so the azure tts voice list does not appear in my app data folder, but the voices do not speak even though they appear in sapi 5 voices.
can you please help me?
thank you.

**Comment by gexgd0419** [comment]
Could you check the log.txt file inside the log folder, and tell me what's inside?

**Comment by fettahpinar** [comment]
Interestingly, there is nothing in the log file.
Now when I disable the sapi 5 integration and download and run the application again, my settings and even the api key I added appeared.
However I completely deleted the natural voice folder in the app data folder.
now nothing shows up in the log file.

**Comment by fettahpinar** [comment]
I'm sorry, I've now found out what the log file says.
[2024-09-09 08:06:48.851] [info] Voice enum: Enumerated 2 voice(s)
[2024-09-09 08:06:48.870] [warning] Cache: Cannot download JSON cache content AzureVoiceListCache.json: resolve: Bilinen böyle bir ana bilgisayar yok.
[2024-09-09 08:06:54.958] [info] Cloud voice (Rest API) created: tr-TR-AhmetNeural
[2024-09-09 08:06:59.659] [info] Cloud voice (Rest API) created: tr-TR-EmelNeural
[2024-09-09 08:07:00.746] [info] Cloud voice (Rest API) created: tr-TR-AhmetNeural
[2024-09-09 08:07:01.834] [info] Cloud voice (Rest API) created: tr-TR-EmelNeural
[2024-09-09 08:07:03.179] [info] Cloud voice (Rest API) created: tr-TR-EmelNeural
[2024-09-09 08:07:03.394] [info] Cloud voice (Rest API) created: tr-TR-AhmetNeural
[2024-09-09 08:07:11.865] [error] Speak: Bilinen böyle bir ana bilgisayar yok.
[2024-09-09 08:07:16.232] [error] Speak: Bilinen böyle bir ana bilgisayar yok.

**Comment by gexgd0419** [comment]
This engine will take the "region" text and append ".tts.speech.microsoft.com" to it to get the host domain name.

For example, if the region is "japaneast", the host domain name will be "japaneast.tts.speech.microsoft.com". You can ping this host, so this host exists and the region is correct.

You can check if your region setting is correct. Make sure it's copied from the "Keys and Endpoint" page, and does not contain extra spaces.

**Comment by fettahpinar** [comment]
my region is set to france central.
then what should I enter in the region section?
Actually, wouldn't it be better if you put the region options in this region section, not the writing field?
What do I need to enter in my region setting right now?

**Comment by fettahpinar** [comment]
https://francecentral.api.cognitive.microsoft.com/

**Comment by gexgd0419** [comment]
Then the region should just be "francecentral".

**Comment by fettahpinar** [comment]
now the problem is completely solved!
I'm really grateful to you!
Thank you very, very much.

**Comment by gexgd0419** [comment]
Usually you will be able to copy everything you need in the `Resource Management > Keys and Endpoint` page of your speech resource. The "region" can be copied from the "Location/Region" text box (in Turkish, it should be called "Konum/Bölge"), so you don't actually need to type it in.

**Comment by fettahpinar** [comment]
Yes, I understand.
I really thank you very much for your interest and concern.
By the way, I have one last question for you.
microsoft has recently released a preview of about 40 multilingual voices.
In the natural voice application, in the language selection screen, after selecting the turkish language, is it possible to see multilingual voices in this voice list?
So let me explain it like this.
a multilingual voice whose native language is italian can also read in turkish.
Instead of selecting all languages and trying to find multilingual sounds from a list of 400 sounds, is it possible that when we select only the Turkish language, both native Turkish sounds and multilingual sounds will appear?

**Comment by gexgd0419** [comment]
This is an interesting idea. Currently it's not possible, but I will consider adding the feature in a future release.

A problem with the multilingual voices is that it may use the wrong language, and speak the text in a weird pronunciation. The issue #2 mentions that problem. An SSML tag `<lang xml:lang="xxx">` can solve this problem for Azure voices (not for Edge voices, unfortunately), but most apps won't do that, so I'll have to fix this on my side.

Maybe I can give users an option to split one Azure multilingual voices into different SAPI voices for each language. For example, `Brian Multilingual` can be splitted into `Brian - English (United States)`, `Brian - French (France)`, `Brian - Turkish (Turkey)` and so on. Users will be able to choose the correct language, instead of letting Azure to guess the language. This will increase the voice count, so only the "sub-voices" that match the included language settings will be shown (you won't get `Brian - French (France)` if you don't select French in the language settings).

Do you think this is a good idea, or do you have a better way?

**Comment by fettahpinar** [comment]
Wow!
this is a really excellent idea, you can do it this way!
the only problem is that the voice won't switch languages, so it will always speak in the same language.
Actually, I sometimes read texts in both Turkish and Arabic and I was thinking of using multilingual voices for this purpose.
but splitting and distributing the multilingual voices is a great idea!

**Comment by fettahpinar** [comment]
yes, it is true that multilingual voices pronounce some sentences in a strange way, I agree with that!
As I said, what I just want to do is to introduce these multilingual sounds in both Turkish and Arabic texts.
but your suggestion makes more sense.
there are a few Turkish voices that really do excellent readings in Turkish and it would be great to listen to texts with them without language confusion!
thank you very much for considering such a suggestion!
really the application you have developed works perfectly!

**Comment by fettahpinar** [comment]
Hello.
when entering and saving the api key and switching to sapi 5 sounds, the current sound list is not downloaded.
even though i have checked the folder many times, it uses its own sound list in the app and the current sound list is not downloaded.
can you please take care of this problem?

**Comment by gexgd0419** [comment]
The fix will be in the next release.

**Comment by gexgd0419** [comment]
This should be fixed in [v0.2.2](https://github.com/gexgd0419/NaturalVoiceSAPIAdapter/releases/tag/v0.2.2).

**Comment by fettahpinar** [comment]
Thank you very much indeed for the correction.
also in these new versions the read all command is now supported and a document can be read from start to finish.
I really thank you again for developing such a tool.
Are you going to work on making all multilingual voices appear in one selected language?

**Comment by gexgd0419** [comment]
> Are you going to work on making all multilingual voices appear in one selected language?

An option to show all multilingual voices has been added in [v0.2.6](https://github.com/gexgd0419/NaturalVoiceSAPIAdapter/releases/tag/v0.2.6).
